### PR TITLE
Add e2e tests to confirm that the store is in coming soon mode after completing the core profiler

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds e2e tests to confirm that the store is in coming soon mode after completing the core profiler to catch any regressions in the future.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Review the changes and ensure e2e CI tests are passing.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add e2e tests to confirm that the store is in coming soon mode after completing the core profiler

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by psealock — APPROVED** [review]
Thanks @chihsuan this is a great use case for e2e tests 🚢

File: plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/core-profiler.spec.js [hunk 1]
```diff
@@ -394,11 +408,33 @@ test.describe( 'Store owner can complete the core profiler', () => {
 			).toBeHidden();
 		} );
 	} );
+
+	test( 'Confirm that the store is in coming soon mode after completing the core profiler', async ( {
+		page,
+	} ) => {
+		await page.goto( 'wp-admin/admin.php?page=wc-admin' );
```
Comment (rjchow) [inline plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/core-profiler.spec.js]: I'd suggest that this assertion be either part of one of the above flows that executes the steps for the completion of the core profiler, or it should itself include the steps for the completion of the core profiler. 

The current way it's written means that it's dependent on the other tests running prior to it and that's not the most ideal test design, as tests should be able to run independently and/or in any order

**Review by rjchow — CHANGES_REQUESTED** [review]
Nice work! Have some feedback for a better way to structure it

File: plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/core-profiler.spec.js (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/core-profiler.spec.js]: Thats a good idea, I needed to read the previous tests to see what was going on

Comment (chihsuan) [reply plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/core-profiler.spec.js]: Great advice! @rjchow Updated the tests to include the coming soon test case as a step in the core profiler completion flow in 54626510187f055203ea3be6a876f877cb4c302e

**Review by rjchow — APPROVED** [review]
thanks! lgtm!

**Pull request merged**